src/main/java/org/opensearch/geospatial/plugin/GeospatialPlugin.java

```diff
@@ -265,4 +275,48 @@ public List<AggregationSpec> getAggregations() {
 
         return List.of(geoHexGridSpec);
     }
+
+    @Override
+    public void onNodeStarted(DiscoveryNode localNode) {
+        LockService lockService = GuiceHolder.getLockService();
+        ip2GeoLockService.initialize(lockService);
+
+        DatasourceRunner.getJobRunnerInstance()
+            .initialize(this.clusterService, this.datasourceUpdateService, this.ip2GeoExecutor, this.datasourceDao, this.ip2GeoLockService);
+    }
+
+    public static class GuiceHolder implements LifecycleComponent {
+
+        private static LockService lockService;
+
+        @Inject
+        public GuiceHolder(final LockService lockService) {
+            GuiceHolder.lockService = lockService;
+        }
+
+        public static LockService getLockService() {
```

Comment:
Might be good to make it as package private scope so that this method is not being used widely in the code.
This static pattern should be restricted to minimize its impact as it make hard to write a unit test.
```suggestion
        static LockService getLockService() {
```